### ISSUE TITLE
Desktop client starts checking for local changes every 30 seconds

### ISSUE THREAD

**Issue by despens** [nextcloud/desktop#2911]
### Expected behaviour
During regular operation the desktop client should get notified of local file system changes and then sync changes to the server. 

### Actual behaviour
The client scans directories every 30 seconds, using lots of CPU on IO on large file collections.

### Steps to reproduce
1. Make directory with lots of files (hundreds of thousands) in many sub directories
2. Set up nextcloud client to sync this directory with server
3. Wait for initial sync to complete, then watch the client looking for 'local changes' and after completing the scan repeating it after a 30 sec pause.

### Client configuration
Client version: 3.1.2 (Ubuntu)

Operating system: Ubuntu 20.,04

OS language: en-us

Qt version used by client package (Linux only, see also Settings dialog): not displayed in settings dialog

Client package (From Nextcloud or distro) (Linux only): `ii  nextcloud-desktop                          3.1.2-20210128.185328.9398260fb-1.0~focal1 amd64        Nextcloud folder synchronization tool`

Installation path of client: /usr/bin/nextcloud

### Server configuration

Nextcloud version: 20.0.6 docker

Storage backend (external storage): s3 object store

### Logs

1. Client logfile: Please advise what log files you need and how to share. The log files contain file names and server URLs that cannot be publicly shared.

2. Web server error log: no errors

**Comment by camilasan** [comment]
Could you try to add the option forceSyncInterval to the config file?
https://docs.nextcloud.com/desktop/3.1/advancedusage.html#configuration-file

Besides that, the server will start notifying the clients with files changes starting on the coming 21 release (see https://github.com/nextcloud/notify_push and https://github.com/nextcloud/desktop/pull/2814).

**Comment by despens** [comment]
The server going to notify the sync client is great news, very much looking forward to version 21!

As for your hint on the config file, I added `forceSyncInterval=7200000`and `remotePollInterval=300000` under `[General]`, but do not notice any change. My large local directories are constantly scanned, and the nextcloud client is using in between 12% to 20% CPU all the time.

**Comment by despens** [comment]
I was eager to install version 21 and the client push server, finished this today on two nextcloud instances.

Using the latest nextcloud desktop client 3.1.3, how can I verify that it indeed uses this new capabilities?

**Comment by FlexW** [comment]
You can just edit a text file on the server and see if the text file changes on your computer nearly immediately. A other way would be to look at log files and look for a string "Push notifications ready".

**Comment by despens** [comment]
The logs are impossible to verify for anything because probably due to the constant checking of local files they're filling up super quick. Overall, my local file trees remain to be checked for changes every 30 seconds.  Two log files with a size of more than 60 MB (uncompressed) are created _per minute_!

Nextcloud desktop continues to put a high load on my local disk and CPU.

Would it be possible to share this log file with a nextcloud desktop client developer directly? I do not want to post them publicly because they contain semi-sensitive information in the file names, and are too large to be properly anonymized thoroughly.

<details>
<summary>List of logfiles</summary>

```
despens@slice:~/.config/Nextcloud/logs$ ls
20210308_2129_owncloud.log.0.gz   20210308_2132_owncloud.log.0.gz   20210308_2133_owncloud.log.9.gz  20210308_2139_owncloud.log.0.gz
20210308_2129_owncloud.log.10.gz  20210308_2132_owncloud.log.1.gz   20210308_2134_owncloud.log.0.gz  20210308_2140_owncloud.log.0.gz
20210308_2129_owncloud.log.1.gz   20210308_2132_owncloud.log.2.gz   20210308_2134_owncloud.log.1.gz  20210308_2140_owncloud.log.1.gz
20210308_2129_owncloud.log.2.gz   20210308_2132_owncloud.log.3.gz   20210308_2134_owncloud.log.2.gz  20210308_2141_owncloud.log.0.gz
20210308_2129_owncloud.log.3.gz   20210308_2132_owncloud.log.4.gz   20210308_2134_owncloud.log.3.gz  20210308_2142_owncloud.log.0.gz
20210308_2129_owncloud.log.4.gz   20210308_2132_owncloud.log.5.gz   20210308_2134_owncloud.log.4.gz  20210308_2142_owncloud.log.1.gz
20210308_2129_owncloud.log.5.gz   20210308_2132_owncloud.log.6.gz   20210308_2134_owncloud.log.5.gz  20210308_2143_owncloud.log.0.gz
20210308_2129_owncloud.log.6.gz   20210308_2133_owncloud.log.0.gz   20210308_2134_owncloud.log.6.gz  20210308_2143_owncloud.log.1.gz
20210308_2129_owncloud.log.7.gz   20210308_2133_owncloud.log.10.gz  20210308_2134_owncloud.log.7.gz  20210308_2143_owncloud.log.2.gz
20210308_2129_owncloud.log.8.gz   20210308_2133_owncloud.log.11.gz  20210308_2134_owncloud.log.8.gz  20210308_2144_owncloud.log.0.gz
20210308_2129_owncloud.log.9.gz   20210308_2133_owncloud.log.1.gz   20210308_2134_owncloud.log.9.gz  20210308_2144_owncloud.log.1.gz
20210308_2130_owncloud.log.0.gz   20210308_2133_owncloud.log.2.gz   20210308_2135_owncloud.log.0.gz  20210308_2145_owncloud.log.0.gz
20210308_2130_owncloud.log.1.gz   20210308_2133_owncloud.log.3.gz   20210308_2135_owncloud.log.1.gz  20210308_2145_owncloud.log.1.gz
20210308_2131_owncloud.log.0.gz   20210308_2133_owncloud.log.4.gz   20210308_2135_owncloud.log.2.gz  20210308_2146_owncloud.log.0.gz
20210308_2131_owncloud.log.1.gz   20210308_2133_owncloud.log.5.gz   20210308_2136_owncloud.log.0.gz  20210308_2146_owncloud.log.1.gz
20210308_2131_owncloud.log.2.gz   20210308_2133_owncloud.log.6.gz   20210308_2136_owncloud.log.1.gz  20210308_2146_owncloud.log.2.gz
20210308_2131_owncloud.log.3.gz   20210308_2133_owncloud.log.7.gz   20210308_2137_owncloud.log.0.gz  20210308_2147_owncloud.log.0.gz
20210308_2131_owncloud.log.4.gz   20210308_2133_owncloud.log.8.gz   20210308_2138_owncloud.log.0.gz  20210308_2147_owncloud.log.1
```

</details>

**Comment by FlexW** [comment]
The log files should not that big. Sounds a little bit similar to https://github.com/nextcloud/desktop/issues/2854. You can drop your logs here https://cloud.nextcloud.com/s/zREzi95CGswYECy.

**Comment by despens** [comment]
Thank you @FlexW, dropped `despens-desktop-issue2911-20210308_2150_owncloud.log.0.gz` as an example. Indeed #2854 looks very similar to my issue.

**Comment by FlexW** [comment]
@despens From this log I can't tell you if the push notifications backend is running. Indeed you have a lot of log messages. I'm unsure why this is the case but you seem to sync a lot of files and have a lot of ignored files. Can you tell me how much files and how big they are in size? We release in the next couple of days a new version of the desktop client. This update contains a major update to the sync engine and the part that is logging so much. Would be great if you could test with that release if you still have that problem.

**Comment by despens** [comment]
Thank you for looking at my log @FlexW! My ignore files list is very small, just 

```console
$ cat sync-exclude.lst 
.Trash-*/
```

What I am making big use of is unchecking directories in the settings. Larger directories that I do not need for current work are disabled for syncing so the master data remains on nextcloud only.

The server interface doesn't show me a statistic (maybe I don't know where to look), but there is 1TB of data and hundreds and thousands of files on my nextcloud account. On my local computer the file numbers currently I keep 46'733 files of overall 9'2 GB under the control of nextcloud desktop. This did not _use_ to be a problem until the client started to check for local changes so excessively. I believe it must have used something like inotify to get notified of local changes, and now that mechanism fails?

Very much looking forward testing with a new client and sync engine.

**Comment by FlexW** [comment]
@despens Thank you for that information. If the problem comes back let me know. I will then try to reproduce it and setup a test environment with equal conditions.

**Comment by despens** [comment]
Quick report: I did setup the metrics monitoring for the client push app on nextcloud server. `active_connection_count 0` is displayed when I'm connected with just the desktop client, `active_connection_count 1` when I run both the desktop client and use the web interface. `total_connection_count 2` always stays the same.

Does this indicate push notifications working or not working with the desktop client?

**Comment by despens** [comment]
After upgrading to 3.2.0, the client produced a helpful error message, which I copied from the log below:

09:52:26||folder_name/notes.md|8|2|1615237035|60468faba0944|7235|06673655octvz01shbpt|2|The downloaded file does not match the checksum, it will be resumed. '71a1d41a72322ff99806569fd78630a448ff3d88' != 'd3471b9d307ea849eb25b2257d628386ecec37f6'|200|5252|1602147143|9e1eac67-b863-43b7-840f-0ce176465f4f|

This error message was repeated every few minutes and apparently caused the frequent sync restart.

The file in question was a collaboratively edited markdown file. Apparently there was an issue when the file contents became out of sync with a checksum?

After making a copy of the file and then deleting the offending original file, the nextcloud desktop client is running smooth as silk. :heart_eyes: 

(It looks like the collaborative markdown editing using the nextcloud text editing app is pretty poor when using external object storage. I checked and noticed the contents of the local file and the nextcloud file were out of sync.)

For me, this new error message fixed the issue, but maybe you want to keep the issue open or make a new one to address the checksum problem.
